pulp_container/app/registry_api.py

```diff
@@ -378,36 +383,136 @@ def get(self, request):
         return Response(data={})
 
 
-class CatalogView(ContainerRegistryApiMixin, APIView):
+class ContainerCatalogSerializer(ModelSerializer):
+    """
+    Serializer for Distributions in the _catalog endpoint of the registry.
+    """
+
+    class Meta:
+        model = models.ContainerDistribution
+        fields = ["base_path"]
+
+
+class ContainerCatalogPagination(BasePagination):
+    """
+    Pagination class to paginate repositories by names according to the registry api specification.
+    """
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """
+        Analyse the pagination parameters and prepare the queryset.
+        """
+        try:
+            self.n = int(request.query_params.get("n"))
+        except Exception:
+            self.n = self.page_size
+        else:
+            if self.n > 10 * self.page_size:
+                self.n = 10 * self.page_size
```

Comment:
For the greedy.
In the end the pagination is not really a service to the user, but a precautionary measure against DOS.

Reply:
i would not know what's the reasonable max page_size, probably 1000 is ok if having 100 as defaults unchanged